PWGHF/Tasks/taskX.cxx

```diff
@@ -66,13 +71,20 @@ struct TaskX {
     registry.add("hDecLenXYErr", "3-prong candidates;decay length xy error (cm);entries", {HistType::kTH2F, {{100, 0., 0.01}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
   }
 
-  Filter filterSelectCandidates = (aod::hf_selcandidate_x::isSelXToJpsiPiPi >= d_selectionFlagX);
+  Filter filterSelectCandidates = (aod::hf_selcandidate_x::isSelXToJpsiPiPi >= d_selectionFlagX || aod::hf_selcandidate_x::isSelXToJpsiToMuMuPiPi >= d_selectionFlagX);
 
   void process(soa::Filtered<soa::Join<aod::HfCandX, aod::HFSelXToJpsiPiPiCandidate>> const& candidates)
   {
+    int decayMode = d_modeXToJpsiToMuMuPiPi ? XToJpsiToMuMuPiPi : XToJpsiPiPi;
     for (auto& candidate : candidates) {
-      if (!(candidate.hfflag() & 1 << XToJpsiPiPi)) {
-        continue;
+      if (d_modeXToJpsiToMuMuPiPi) {
+        if (!(candidate.hfflag() & 1 << decayMode)) {
+          continue;
+        }
+      } else {
+        if (!(candidate.hfflag() & 1 << decayMode)) {
+          continue;
+        }
```

Comment:
The code looks identical in both cases. I would say that the `if`/`else` switch is not needed.

Reply:
Done